### PR TITLE
XRT-1949: Added Zigbee Example

### PR DESCRIPTION
**Pull request by AIM289**

### CODE REVIEW

**Review by FraserG01 — CHANGES_REQUESTED** [review]
- Require getting started sub-section on the device that is being used
- Require getting started sub-section for running zigbee2mqtt, also provide a start_zigbee2mqtt.sh and stop_zigbee2mqtt.sh commands
-

**Comment by AIM289** [comment]
> * Require getting started sub-section on the device that is being used
> * Require getting started sub-section for running zigbee2mqtt, also provide a start_zigbee2mqtt.sh and stop_zigbee2mqtt.sh commands
> * Also require extra section on readme on protocol specific commands, see: https://github.com/IOTechSystems/xrt-examples/tree/XRT-1949-branch/DeviceServices/opc-ua#subscriptions and https://github.com/IOTechSystems/xrt-examples/tree/XRT-1949-branch/DeviceServices/bacnet-ip#change-of-value-subscriptions for examples

Added these sections as well as z2mqtt start/stop scripts

File: DeviceServices/zigbee/commands/add_polled_schedule.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+mosquitto_pub -t xrt/devices/zigbee/request -m \
```
Comment (FraserG01) [inline DeviceServices/zigbee/commands/add_polled_schedule.sh:3]: file should be named add_schedule as the docs refer to the command as that

File: DeviceServices/zigbee/commands/remove_polled_schedule.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+mosquitto_pub -t xrt/devices/zigbee/request -m \
```
Comment (FraserG01) [inline DeviceServices/zigbee/commands/remove_polled_schedule.sh:3]: file should be named remove_schedule as the docs refer to the command as that

File: DeviceServices/zigbee/deployment/config/main.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "logger":"IOT::Logger",
+  "logger_debug":"IOT::Logger",
```
Comment (FraserG01) [inline DeviceServices/zigbee/deployment/config/main.json:3]: dont need logger_debug

File: DeviceServices/zigbee/deployment/config/logger_debug.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (FraserG01) [inline DeviceServices/zigbee/deployment/config/logger_debug.json:1]: done need logger_debug

File: DeviceServices/zigbee/deployment/state/schedules.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"apiVersion":"v2","schedules":[]}
```
Comment (FraserG01) [inline DeviceServices/zigbee/deployment/state/schedules.json:1]: there should be an initial schedule

File: DeviceServices/zigbee/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+This page shows you how to set up and run the Zigbee Device Service example.
+
+For more information about the Device Service please review the [Zigbee Device Service](https://docs.iotechsys.com/edge-xrt21/device-service-components/zigbee-device-service-component.html) documentation.
+
+## Getting Started
+
+
+### Example Devices
+
+Unlike many of the examples for the other Device Services, there is no readily available Zigbee device simulator. Therefore, this example is instead based off two real world devices which are required to run the example. The devices used are an [OSRAM Flex RGBW LED Light Strip](https://www.zigbee2mqtt.io/devices/4052899926110.html) and an [Immax Intelligent Motion Sensor](https://www.zigbee2mqtt.io/devices/07047L.html). Click on each device to view their respective pages within the Zigbee2MQTT documentation.
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:14]: worth having a note saying the friendly names would need to be updated for this example to actually work?

Could also put a note that if using different devices they could just clear devices.json and schedules.json and run discovery with AutoRegister turned to true?

File: DeviceServices/zigbee/commands/start_zigbee2mqtt.sh [hunk 7]
```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+read -p "Input Zigbee adapter location:" adapterlocation
```
Comment (FraserG01) [inline DeviceServices/zigbee/commands/start_zigbee2mqtt.sh:3]: Arguments would be better to use as can be avoids runtime user input

File: DeviceServices/zigbee/README.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+### Start Zigbee2MQTT
+
+*For more information on Zigbee2MQTT, please read the [Zigbee2MQTT documentation](https://www.zigbee2mqtt.io/)*
+
+The following script simplifies the process of starting Zigbee2MQTT. To proceed, the Zigbee adapter location and the MQTT broker address are required. 
+
+Run:
+
+```bash
+. ./commands/start_zigbee2mqtt.sh
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:25]: is there a reason this shell script needs the first dot

File: DeviceServices/zigbee/README.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+### Stop Zigbee2MQTT
+
+To stop Zigbee2MQTT, run the following:
+
+```bash
+. ./commands/stop_zigbee2mqtt.sh
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:33]: is there a reason this shell script needs the first dot

**Review by FraserG01 — CHANGES_REQUESTED** [review]

File: DeviceServices/zigbee/commands/start_zigbee2mqtt.sh [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  # MQTT server authentication, uncomment if required:
+  # user: my_user
+  # password: my_password
+# Serial settings
+serial:
+  # Location of Zigbee adapter
+  port: $adapterlocation" > data/configuration.yaml
+fi
+
+docker run \
```
Comment (FraserG01) [inline DeviceServices/zigbee/commands/start_zigbee2mqtt.sh:37]: can you add --rm to this run command

File: DeviceServices/zigbee/commands/stop_zigbee2mqtt.sh [hunk 11]
```diff
@@ -0,0 +1,2 @@
+docker stop zigbee2mqtt
+docker rm zigbee2mqtt
```
Comment (FraserG01) [inline DeviceServices/zigbee/commands/stop_zigbee2mqtt.sh:2]: with --rm added to the docker run, the image shouldnt need to be removed

File: DeviceServices/zigbee/commands/add_polled_schedule.sh (hunk 1 above)
Comment (AIM289) [reply DeviceServices/zigbee/commands/add_polled_schedule.sh]: renamed

File: DeviceServices/zigbee/deployment/config/main.json (hunk 3 above)
Comment (AIM289) [reply DeviceServices/zigbee/deployment/config/main.json]: removed

File: DeviceServices/zigbee/deployment/config/logger_debug.json (hunk 4 above)
Comment (AIM289) [reply DeviceServices/zigbee/deployment/config/logger_debug.json]: removed

File: DeviceServices/zigbee/deployment/state/schedules.json (hunk 5 above)
Comment (AIM289) [reply DeviceServices/zigbee/deployment/state/schedules.json]: added

File: DeviceServices/zigbee/README.md (hunk 6 above)
Comment (AIM289) [reply DeviceServices/zigbee/README.md:14]: Added notes

File: DeviceServices/zigbee/commands/start_zigbee2mqtt.sh (hunk 7 above)
Comment (AIM289) [reply DeviceServices/zigbee/commands/start_zigbee2mqtt.sh]: switched to use args

File: DeviceServices/zigbee/README.md (hunk 8 above)
Comment (AIM289) [reply DeviceServices/zigbee/README.md]: removed dot

File: DeviceServices/zigbee/README.md (hunk 9 above)
Comment (AIM289) [reply DeviceServices/zigbee/README.md]: removed dot

File: DeviceServices/zigbee/commands/start_zigbee2mqtt.sh (hunk 10 above)
Comment (AIM289) [reply DeviceServices/zigbee/commands/start_zigbee2mqtt.sh:33]: added

File: DeviceServices/zigbee/commands/stop_zigbee2mqtt.sh (hunk 11 above)
Comment (AIM289) [reply DeviceServices/zigbee/commands/stop_zigbee2mqtt.sh]: removed

File: DeviceServices/zigbee/README.md [hunk 12]
```diff
@@ -22,15 +25,20 @@ The following script simplifies the process of starting Zigbee2MQTT. To proceed,
 Run:
 
 ```bash
-. ./commands/start_zigbee2mqtt.sh
+./commands/start_zigbee2mqtt.sh arg1 arg2 arg3
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:30]: not better having labelled arguments?

--adapter=<location>

etc

Comment (AIM289) [reply DeviceServices/zigbee/README.md]: Now uses labelled args

File: DeviceServices/zigbee/README.md [hunk 13]
```diff
@@ -27,13 +27,13 @@ The following script simplifies the process of starting Zigbee2MQTT. To proceed,
...
+./commands/start_zigbee2mqtt.sh --adapter=ADAPTER_LOCATION --network=MQTT_NETWORK --broker_address=MQTT_ADDRESS
 ```
-Using the following arguments:
+Using the following values:
 
-- arg1: Location of Zigbee adapter
-- arg2: Whether MQTT broker is on host network (y/n)
-- arg3: Address of MQTT broker
+- ADAPTER_LOCATION: Location of Zigbee adapter
+- MQTT_NETWORK: Network of MQTT broker
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:35]: would it not be better to indicate that is is the the docker network you are choosing (host or the name of a docker network its to run on)

File: DeviceServices/zigbee/README.md [hunk 14]
```diff
@@ -27,13 +27,13 @@ The following script simplifies the process of starting Zigbee2MQTT. To proceed,
 Run:
 
 ```bash
-./commands/start_zigbee2mqtt.sh arg1 arg2 arg3
+./commands/start_zigbee2mqtt.sh --adapter=ADAPTER_LOCATION --network=MQTT_NETWORK --broker_address=MQTT_ADDRESS
 ```
-Using the following arguments:
+Using the following values:
```
Comment (FraserG01) [inline DeviceServices/zigbee/README.md:32]: Worth having an example usage of the full command with filled in arguments?

**Review by FraserG01 — CHANGES_REQUESTED** [review]

File: DeviceServices/zigbee/README.md (hunk 13 above)
Comment (AIM289) [reply DeviceServices/zigbee/README.md]: Changed

File: DeviceServices/zigbee/README.md (hunk 14 above)
Comment (AIM289) [reply DeviceServices/zigbee/README.md:32]: Added example

**Review by FraserG01 — APPROVED** [review]
LGTM

**Pull request merged**